src/app/wallets/get-on-chain-fee.ts

```diff
@@ -34,10 +41,19 @@ export const getOnChainFee = async ({
   const wallet = await walletsRepo.findById(walletIdChecked)
   if (wallet instanceof Error) return wallet
 
-  const withdrawFeeCalculator = WithdrawalFeeCalculator()
+  const minBankFee = toSats(account.withdrawFee)
+
+  const withdrawFeeCalculator = WithdrawalFeeCalculator({
+    feeRatio: getFeesConfig().withdrawRatio,
+    thresholdImbalance: getFeesConfig().withdrawThreshold,
+  })
+
   const payeeWallet = await walletsRepo.findByAddress(address)
+  if (payeeWallet instanceof UnknownRepositoryError) return payeeWallet
 
-  const isIntraLedger = !(payeeWallet instanceof Error)
+  const isIntraLedger = !(
```

Comment:
if you are doing validation here with specific error (although specific error validation is not needed)  then line 59 is not necessary.